### DISCUSSION TITLE
Some Keys do not raise the TextBox.KeyDown event

### DISCUSSION THREAD

**Discussion by cnelson04-beckman** [AvaloniaUI/Avalonia#7844] [Ideas]
Currently the KeyDown event for the TextBox is not raised when the Key.Back, Key.Delete, Key.Right, Key.Left keys are pressed. Not too bad for the Key.Left and Key.Right, but it means that cannot capture these key press events before the value in the Text property has been changed. I have been working on a mask behavior and it is really rather inconvenient to respond to these events programmatically.

**Comment by maxkatz6** [comment]
This events most likely are already handled. Probably for cursor navigation in the text box.
Use dev tools events tab to see which events are handled.

You can handle events before anything else handled them:
```
textBox.AddHandler(KeyDownEvent, OnKeyDown, RoutingStrategies.Tunnel);
```
